precompile/fee_config_manager.go

```diff
@@ -0,0 +1,320 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package precompile
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/commontype"
+	"github.com/ava-labs/subnet-evm/vmerrs"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	minFeeConfigFieldKey = iota + 1
+	// add new fields below this
+	// must preserve order of these fields
+	gasLimitKey = iota
+	targetBlockRateKey
+	minBaseFeeKey
+	targetGasKey
+	baseFeeChangeDenominatorKey
+	minBlockGasCostKey
+	maxBlockGasCostKey
+	blockGasCostStepKey
+	// add new fields above this
+	numFeeConfigField = iota - 1
+)
+
+var (
+	_ StatefulPrecompileConfig = &FeeConfigManagerConfig{}
+
+	// Singleton StatefulPrecompiledContract for setting fee configs by permissioned callers.
+	FeeConfigManagerPrecompile StatefulPrecompiledContract = createFeeConfigManagerPrecompile(FeeConfigManagerAddress)
+
+	setFeeConfigSignature              = CalculateFunctionSelector("setFeeConfig(uint256,uint256,uint256,uint256,uint256,uint256,uint256,uint256)")
+	getFeeConfigSignature              = CalculateFunctionSelector("getFeeConfig()")
+	getFeeConfigLastChangedAtSignature = CalculateFunctionSelector("getFeeConfigLastChangedAt()")
+
+	// 8 fields in FeeConfig struct
+	feeConfigInputLen = common.HashLength * numFeeConfigField
+
+	feeConfigLastChangedAtKey = common.Hash{'l', 'c', 'a'}
+
+	ErrCannotChangeFee = errors.New("non-enabled cannot change fee config")
+)
+
+// FeeConfigManagerConfig wraps [AllowListConfig] and uses it to implement the StatefulPrecompileConfig
+// interface while adding in the contract deployer specific precompile address.
+type FeeConfigManagerConfig struct {
+	AllowListConfig
+	commontype.FeeConfig
+}
```

Comment:
For the other precompiles we simply directly embed the allow list config since it's the only value present, but here it looks like we're embedding JSON fields from both the allow list and fee config together, which is a little bit weird.

Can we add JSON tags to at least the `FeeConfig` field here? I think it could go either way on if the `AllowListConfig` here should have its own JSON tag as well.